src/shapes/Textbox.ts

```diff
@@ -187,11 +188,14 @@ export class Textbox extends IText {
    * @param {Number} charIndex
    * @private
    */
-  _getStyleDeclaration(lineIndex: number, charIndex: number) {
+  protected _getStyleDeclaration(
+    lineIndex: number,
+    charIndex: number
+  ): TextStyleDeclaration {
     if (this._styleMap && !this.isWrapping) {
       const map = this._styleMap[lineIndex];
       if (!map) {
-        return null;
+        return {};
       }
       lineIndex = map.line;
       charIndex = map.offset + charIndex;
```

Comment:
though this is not related I would like to say that this piece of logic has a lot more usages than just in style.
It maps between a wrapped position to a non wrapped position.
It is confusing when to use the wrapped/non wrapped position when working on textbox, error prone and not friendly.
I would have a method that returns the offset give a line, char, isWrapped since it is not changed by wrapping (I hope there is no edge case for that) and use that across all methods as a standard.
This will make working on Text and Textbox easier or else we must keep in mind this constant difference that is applied at a certain point in the lifecycle. e.g. measuring uses to non wrapped, rendering uses the wrapped.
All should us the offset.

Reply:
as i said there are so many things that have been bolted on top of normal text, and this stylemap was one.
This works, is confusing, but it works,